packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -58,13 +64,71 @@ export async function fetchSnapshot(
     );
 }
 
+export async function redeemSharingLink(
+    odspResolvedUrl: IOdspResolvedUrl,
+    storageTokenFetcher: (options: TokenFetchOptions, name: string) => Promise<string | null>,
+) {
+    await getWithRetryForTokenRefresh(async (tokenFetchOptions) => {
+        assert(odspResolvedUrl.sharingLinkToRedeem !== undefined, "Share link should be present");
+        const storageToken = await storageTokenFetcher(tokenFetchOptions, "TreesLatest");
+        const shareId = encodeShareUrl(odspResolvedUrl.sharingLinkToRedeem);
+        assert(shareId !== undefined, "ShareId should be present for share link");
+        const redeemUrl = `${odspResolvedUrl.siteUrl}/_api/v2.0/shares/${shareId}`;
+        const { url, headers } = getUrlAndHeadersWithAuth(redeemUrl, storageToken);
+        headers.prefer = "redeemSharingLink";
+        return fetchAndParseAsJSONHelper(url, { headers });
+    });
+}
+
+export async function fetchSnapshotWithRedeem(
+    odspResolvedUrl: IOdspResolvedUrl,
+    storageTokenFetcher: (options: TokenFetchOptions, name: string) => Promise<string | null>,
+    snapshotOptions: ISnapshotOptions | undefined,
+    logger: ITelemetryLogger,
+    snapshotDownloader: (url: string, fetchOptions: {[index: string]: any}) => Promise<IOdspResponse<IOdspSnapshot>>,
+    putInCache: (valueWithEpoch: IVersionedValueWithEpoch) => Promise<void>,
+): Promise<ISnapshotCacheValue> {
+    let odspSnapshot: ISnapshotCacheValue;
```

Comment:
You can remove odspSnapshot variable by simply returning results in-place

Reply:
yeah, just to avoid 2 returns, I did it this way.